demos/adjoint/adjoint_demo.py

```diff
@@ -0,0 +1,348 @@
+# Adjoint Inverse Demo
+# ================================
+#
+# Introduction
+# ------------
+# In this tutorial, we will demonstrate how to perform an inversion to recover the initial temperature field of an idealised mantle convection simulation using G-ADOPT.
+#
+# The tutorial involves a *twin experiment*, where we assess the performance of the inversion scheme by inverting the initial state of a synthetic reference simulation, known as the "*Reference Twin*". To create this reference twin, we run a forward mantle convection simulation and record all relevant fields (velocity and temperature) at each time step.
+#
+# We have pre-run this simulation by running `forward.py`, and stored model output as a checkpoint file on our servers. These fields serve as benchmarks for evaluating our inverse problem's performance. To download the reference benchmark checkpoint file, click
+# [here](https://data.gadopt.org/demos/adjoint-demo-checkpoint-state.h5"), or alternatively, execute the following command in a terminal:
```

Comment:
Note clicking doesn't work for me. I get:
<Error>
<Code>NoSuchKey</Code>
<Message/>
<BucketName>gadopt</BucketName>
<RequestId>tx000003f30271967ee1dfb-00666157b6-dbcff29-syd1a</RequestId>
<HostId>dbcff29-syd1a-syd1-zg01</HostId>
</Error>

Reply:
A typing issue! Corrected. now